src/hivpy/pregnancy.py

```diff
@@ -0,0 +1,268 @@
+import importlib.resources
+from datetime import timedelta
+from math import pow
+
+import numpy as np
+
+import hivpy.column_names as col
+
+from .common import SexType, rng
+from .pregnancy_data import PregnancyData
+
+
+class PregnancyModule:
+
+    def __init__(self, **kwargs):
+
+        # init pregnancy data
+        with importlib.resources.path("hivpy.data", "pregnancy.yaml") as data_path:
+            self.p_data = PregnancyData(data_path)
+
+        self.can_be_pregnant = self.p_data.can_be_pregnant
+        self.rate_want_no_children = self.p_data.rate_want_no_children  # dependent on time step length
+        self.date_pmtct = self.p_data.date_pmtct
+        self.pmtct_inc_rate = self.p_data.pmtct_inc_rate
+        self.fold_preg = self.p_data.fold_preg
+        self.inc_cat = self.p_data.inc_cat.sample()
```

Comment:
I think this highlights an issue with the code at the moment. `inc_cat` already exists in `DemographicsModule.params` so we need to find a way to expose the data modules, or the modules themselves, to one another. At the moment the only way to do so is through the population object, which would mean passing the population or the demographics module into the constructor of this module.